src/plugins/utils/js-error-utils.ts

```diff
@@ -56,11 +58,39 @@ const appendErrorPrimitiveDetails = (
     rumEvent.message = error.toString();
 };
 
-const appendErrorObjectDetails = (
+const appendErrorSourceMaps = async (
+    error: globalThis.Error,
+    fetchFunction?: SourceMapsFetchFunction
+) => {
+    if (error && error.stack) {
+        try {
+            const stackFrames = await StackTrace.fromError(error, {
+                ajax: fetchFunction
```

Comment:
Issue: What does this ajax option do? I don't see it in documentation. 

```
    export interface StackTraceOptions {
        filter?: (stackFrame: StackFrame) => boolean;
        sourceCache?: SourceCache;
        offline?: boolean;
    }
```

https://github.com/stacktracejs/stacktrace.js/blob/710bba1118d396466ee342a30b3dfd19ecbda8b5/stacktrace-js.d.ts#L12-L16

Reply:
Question: assuming `ajax` option works as intended, does that mean an http request is made on every single JSError? If so, is the result cached?

Reply:
Question: how does `stacktrace-js` identify local source maps?

Reply:
stacktrace-js uses stacktrace-gps to make the fetch calls. The options object is passed directly to `new StackTraceGPS()`, https://github.com/stacktracejs/stacktrace.js/blob/master/stacktrace.js#L104 stacktrace-gps takes `ajax` as an option, https://github.com/stacktracejs/stacktrace-gps/blob/master/README.md

From the stacktrace-gps documentation, 
>ajax: Function (String URL => Promise(responseText)) - Function to be used for making X-Domain requests

This is an optional fetch function that can be configured to send credentials or other headers when making requests for the js and map files.

When an error is processed fetch calls for js files are made for all the urls listed in the error stack, at the end of the file there should be a link to a map file, this is also fetched. I did not notice refetching of these files again by the browser if the same error occurs.